algorithms/src/snark/varuna/varuna.rs

```diff
@@ -676,7 +676,8 @@ where
                     .iter()
                     .map(|input| {
                         let input = input.borrow().to_field_elements().unwrap();
-                        let mut new_input = vec![E::Fr::one()];
+                        let mut new_input = Vec::with_capacity(input.len().max(input_domain.size()));
```

Comment:
as this tries to skip the allocs, it should allocate for the `1 + input.len()`

Reply:
Can you open a new PR for this?

Reply:
> as this tries to skip the allocs, it should allocate for the 1 + input.len()

actually, this should be `(1 + input.len()).max(input_domain.size())`, as there's no telling whether resize will increase or decrease the length